File: 020 - Factorial digit sum/20.c
#include <stdio.h>
main(){
    int i, h = 0, sum = 0, a[500]= {0};

    a[0]=1;

    for (i = 1; i <= 100; i++){
        int j;
        int rem = 0;
        int p = 1;

        for (j = 0;  j <= h; j++){
            p = (a[j]*i) + rem;
            a[j] = p % 10;
            rem = p / 10;

            if ( ( rem != 0 ) && ( j == h ) )
                h++;        //Count number of digits in the number
        }
    }

    for (i = 0; i <= h; i++)
        sum += a[i];        //Calculate sum of the digits

    printf("Sum of the digits of the number 100! = %d\n", sum);

}

File: 020 - Factorial digit sum/20.java

import java.math.BigInteger;
import java.util.ArrayList;
import java.util.List;

class A {

    static List<String> l = new ArrayList<String>();
    static int in = 0;
    static BigInteger i = BigInteger.ONE;

    public static void main(String[] args) {

        BigInteger a = (BigInteger.TEN).multiply(BigInteger.TEN);
        System.out.println(a);

        for (int j = 1; j < 101; j++) {
            i = i.multiply(a);
            a = a.subtract(BigInteger.ONE);
            System.out.println(a);
        }

        System.out.println("100! = " + i);

        String p = i + "";

        for (String r : p.split("", p.length())) {
            l.add(r);
        }

        for (int j = 1; j < l.size(); j++) {
            in = in + Integer.parseInt(l.get(j));
        }

        System.out.println("Sum of the digits in the number 100! = " + in);

    }
}